coda/coda_mdstore/tests/test_views.py

```diff
@@ -0,0 +1,780 @@
+        request = rf.get(url)
+        response1 = views.externalIdentifierSearch(request, ext_id.value)
+
+        # URL with the ark id as a query paramater.
+        url = reverse('coda_mdstore.views.externalIdentifierSearch')
+        request = rf.get(url, {'ark': 'metadc000001'})
+        response2 = views.externalIdentifierSearch(request)
+
+        assert response1.content == response2.content
+        assert response1['Content-Type'] == response2['Content-Type']
+
+    def test_request_with_long_ark_parameter(self, rf):
+        bag = FullBagFactory.create()
+        ext_id = ExternalIdentifierFactory.create(
+            belong_to_bag=bag,
+            value='ark:/67531/metadc000001'
+        )
+
+        # Just like in the previous test, feed the real url to the request
+        # factory because the request path is used in the response content.
+        url = reverse(
+            'coda_mdstore.views.externalIdentifierSearch', args=[ext_id.value])
+        request = rf.get(url)
+        response1 = views.externalIdentifierSearch(request, ext_id.value)
+
+        # URL with the ark id as a query paramater.
+        url = reverse('coda_mdstore.views.externalIdentifierSearch')
+        request = rf.get(url, {'ark': 'ark:/67531/metadc000001'})
+        response2 = views.externalIdentifierSearch(request)
+
+        assert response1.content == response2.content
+        assert response1['Content-Type'] == response2['Content-Type']
+
+
+class TestExternalIdentiferSearchJSON:
+
+    @pytest.mark.xfail(reason='Exception raised without the `ark` query'
+                              ' parameter.')
+    def test_with_without_ark_parameter(self, rf):
+        request = rf.get('/')
+        response = views.externalIdentifierSearchJSON(request)
+        assert response.content == '[]'
+
+    def test_with_invalid_ark_id(self, rf):
+        request = rf.get('/', {'ark': 'ark:/67351/metadc000001'})
+        response = views.externalIdentifierSearchJSON(request)
+        assert response.content == '[]'
+        assert response['Content-Type'] == 'application/json'
+
+    def test_with_valid_ark_id(self, rf):
+        bag = FullBagFactory.create()
+        ext_id = ExternalIdentifierFactory.create(
+            belong_to_bag=bag,
+            value='ark:/67531/metadc000001'
+        )
+
+        request = rf.get('/', {'ark': ext_id.value})
+        response = views.externalIdentifierSearchJSON(request)
+        content = json.loads(response.content)
+
+        assert len(content) == 1
+        assert 'bagging_date' in content[0]
+        assert content[0]['name'] == bag.name
+        assert content[0]['oxum'] == '{0}.{1}'.format(bag.size, bag.files)
+
+    def test_with_with_multiple_identifiers(self, rf):
+        bag = FullBagFactory.create()
+        ext_id = ExternalIdentifierFactory.create(
+            belong_to_bag=bag,
+            value='ark:/67531/metadc000001'
+        )
+        ExternalIdentifierFactory.create(
```

Comment:
yeah, this tests the outcome if there are two identifiers with the same `value`. Only one of the instances need to be kept track of in order though.